state/device_data_table.go

```diff
@@ -25,14 +25,15 @@ type DeviceDataTable struct {
 
 func NewDeviceDataTable(db *sqlx.DB) *DeviceDataTable {
 	db.MustExec(`
-	CREATE SEQUENCE IF NOT EXISTS syncv3_device_data_seq;
 	CREATE TABLE IF NOT EXISTS syncv3_device_data (
-		id BIGINT PRIMARY KEY NOT NULL DEFAULT nextval('syncv3_device_data_seq'),
 		user_id TEXT NOT NULL,
 		device_id TEXT NOT NULL,
 		data BYTEA NOT NULL,
 		UNIQUE(user_id, device_id)
 	);
+	-- Set the fillfactor to 90%, to allow for HOT updates (e.g. we only
+	-- change the data, not anything indexed like the id)
+	ALTER TABLE syncv3_device_data SET (fillfactor = 90);
```

Comment:
Isn't this redundant now we're removing the `id` column?

Reply:
We need both, as still every `UPDATE` results in the `ctid` ("current tuple ID") to be updated by Postgres. While it is possible that there is space in the "current block" (because we "cleared" space by a different update), this raises the possibility for it.